### PR TITLE
Treat rustfmt as optional

### PR DESCRIPTION
**Pull request by illicitonion**
All the code above this call treats rustfmt as optional, but then when
we make the BUILD for the rust toolchain, we unconditionally point at
the presumed rustfmt label.

At
https://github.com/bazelbuild/rules_rust/blob/59fab4e79f62bfa13551ac851a40696c24c0c3a4/rust/toolchain.bzl#L331-L335
we conditionally create a symlink based on whether the attribute is set,
but because we unconditionally set the attribute, this causes a hard
error failing to symlink to a non-existent file.

Instead, treat rustfmt as optional in this function too, so that the
code above and below properly convey the optionality to each other.

### CODE REVIEW

File: rust/private/repository_utils.bzl [hunk 1]
```diff
@@ -235,6 +236,9 @@ def BUILD_for_rust_toolchain(
     rustc_srcs = "None"
     if include_rustc_srcs:
         rustc_srcs = "\"@{workspace_name}//lib/rustlib/src:rustc_srcs\"".format(workspace_name = workspace_name)
+    rustfmt_label = "None"
+    if rustfmt_version:
+        rustfmt_label = "@{workspace_name}//:rustfmt_bin".format(workspace_name = workspace_name)
```
Comment (UebelAndre) [inline rust/private/repository_utils.bzl:241]: You'll need quotes here if you want to be able to pass `None` in the template in this way.
```suggestion
        rustfmt_label = "\"@{workspace_name}//:rustfmt_bin\"".format(workspace_name = workspace_name)
```

**Review by UebelAndre — CHANGES_REQUESTED** [review]

File: rust/private/repository_utils.bzl [hunk 2]
```diff
@@ -210,6 +210,7 @@ def BUILD_for_rust_toolchain(
         target_triple,
         include_rustc_srcs,
         default_edition,
+        rustfmt_version,
```
Comment (UebelAndre) [inline rust/private/repository_utils.bzl:213]: nit: since the version is not actually used, can this be something like `include_rustfmt`?

**Review by UebelAndre — CHANGES_REQUESTED** [review]
Looks good! Just one request 😄

File: rust/private/repository_utils.bzl (hunk 2 above)
Comment (illicitonion) [reply rust/private/repository_utils.bzl]: Changed to `has_rustfmt` :)

**Pull request merged**